internal/controller/ols_app_server_reconciliator.go

```diff
@@ -240,3 +245,34 @@ func (r *OLSConfigReconciler) reconcileService(ctx context.Context, cr *olsv1alp
 	r.logger.Info("OLS service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileTLSSecret(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	var secretValues map[string]string
+	foundSecret := &corev1.Secret{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: OLSCertsSecretName, Namespace: r.Options.Namespace}, foundSecret)
+	if err != nil {
+		return fmt.Errorf("failed to fetch secret: %w", err)
```

Comment:
@vishnuchalla - Can we update this error message - This is caused most likely due the tls-secret not present.